src/gui/sdlmain.cpp

```diff
@@ -1460,21 +1460,65 @@ RenderingBackend GFX_GetRenderingBackend()
 	return sdl.rendering_backend;
 }
 
+static SDL_Rect get_desktop_size()
+{
+	SDL_Rect desktop;
+	assert(sdl.display_number >= 0);
+
+	SDL_GetDisplayBounds(sdl.display_number, &desktop);
```

Comment:
Possibly uninitialized variable `SDL_Rect desktop;`.  It's not initialized to 0 and the return value of this function is not checked.

Also, this entire function seems strange to me.  Why do we care about the size of window decorations when calculating desktop size?

Reply:
Dunno, but out of scope for now 😅 I'll add the defaulting, though.

Reply:
> Why do we care about the size of window decorations when calculating desktop size?

In windowed mode, if you set `windowresolution = desktop`, SDL will avoid clobbering/overlapping your main control panel (similar to setting an application to "maximized" window state).  SDL is smart enough to be able to query the WM on Windows, macOS, Linux, (even BSD), and also works if the desktop OS supports using a vertical-side panel instead of at the top or bottom.

Reply:
That was my guess, thanks for clarifying @kcgen .